### ISSUE TITLE
[2017 ICLR Oral] Understanding Deep Learning requires Rethinking Generalization

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#70]
2017 ICLR best paper award를 받은 논문

기존의 traditional approach로 neural network가 왜 generalization이 잘되는지에 대하여 해석할 수 없음을 보임

논문의 주장에 대한 실험과 간단한 의견만 요약

### 1.1. Fitting Random Labels and Pixels
<img width="1172" alt="image" src="https://user-images.githubusercontent.com/49643709/169202269-466843f1-94c4-48b7-be47-08b682725a14.png">
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/49643709/169202308-a5a2c05d-116e-4c9d-9162-ff5489aff748.png">

Neural network는 random labels, shuffled pixels, random pixels, gaussian noise 등에 대하여 충분히 학습할 수 있음

Input과 label의 correlation이 없기에, 해당 실험이 의미하는 바는 neural network는 dataset을 충분히 memorize할 수 있다는 말임

### 1.2. Implications

- Rademacher complexity and VC-dimension
    - 모델이 random label에 대해 학습할 수 있는지에 대한 정도를 이용하여 capacity를 측정
    - 해당 capacity가 크면 generalize 능력이 떨어짐
    - Neural network는 random label에 대해 학습할 수 있는데 generalize 잘하기에 안맞음
- Uniform stability
    - 알고리즘이 single example의 replacement에 얼마나 sensitive한지를 측정
    - 알고리즘이 sensitive하면 generalize 능력이 떨어짐
    - 이는 data, label distribution과 관계없이 알고리즘의 property만 보기에 적합하지 않음

즉 traditional approaches들로 neural network의 generalization을 설명할 수 없음

### 2.1. Explicit Regularization
<img width="1232" alt="image" src="https://user-images.githubusercontent.com/49643709/169255046-986565e8-336e-4e57-b562-cef17177f7dc.png">
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/49643709/169254908-d0a870f0-0f1a-4fba-8e14-5e0789fc163c.png">

Randomization test를 진행할 때, explicit regularization을 적용하지 않았음

Parameters가 data points보다 많은 경우, overfitting을 방지하기 위해 regularization을 사용함
(regularizers help confine learning to a subset of the hypothesis space with manageable complexity)

Randomization test에서 neural network에 explicit regularization을 적용하여 학습해도 random training set에 충분히 학습이 됨
(however, AlexNet with weight decay fails to converge on random labels)

또한 explicit regularization이 있으면 generalization 성능이 높아지지만 없더라도 neural network는 충분히 generalize 잘함

이로부터 neural network에서 explicit regularization은 기존과 다른 역할을 한다고 볼 수 있음

### 2.2. Implicit Regularization
<img width="1235" alt="image" src="https://user-images.githubusercontent.com/49643709/169257494-1a0a3292-191c-4878-862f-3e8adb57448c.png">

ImageNet에서는 early stopping을 하면 generalize performance를 높일 수 있지만 CIFAR10 dataset에서는 소용 없음

Batch normalization을 사용하면 learning dynamics를 충분히 stabilize할 수 있지만, performance의 generalization impact는 3~4% 밖에 차이나지 않음

결국 regularization을 사용하면 generalization performance를 높일 수 있는 것은 맞지만, regularization 없이도 network가 충분히 generalization 능력이 좋기에 generalization의 fundamental reason이 될 수 없음

## 3. Implicit Regularization : An Appeal to Linear Models

Neural network는 복잡하여 이해가 어렵기에, 논문에서는 보다 간단한 linear model로 실험

<img width="799" alt="image" src="https://user-images.githubusercontent.com/49643709/169335342-ece57351-c476-42d5-8e77-322ebd3a2469.png">

저자들은 Theorem 1을 증명하여 2-layer neural network로 충분히 data를 memorize를 할 수 있는 capacity가 됨을 보임

Input data의 수가 n, input data의 dimension이 d일때 d >= n이면 어떠한 label이던 fitting할 수 있음

Input data의 dimension이 input data의 수보다 많다면, 모델은 여러 global minimum을 가질 수 있음

어떤 global minimum이 더 generalization 능력이 좋은지를 평가하는 방법으로 minima의 loss function의 curvature를 계산할 수 있음

하지만 linear model의 curvature은 모든 optima에서 같기에, 이를 가지고 구분할 수 없음

SGD가 어떻게 weight update를 하고 converge를 하는지 보면, minimum l2-norm solution과 동일한 의미를 가지고 있음

즉 SGD는 solution이 minimum norm solution을 가지도록 converge함

Random label을 학습한 모델의 norm이 true label을 학습한 모델의 norm보다 크다는 것을 확인할 수 있었음

하지만 실험을 통해 minimum norm을 가진다고해서 generalization performance가 더 좋은지 예측할 수 없었음
(MNIST dataset에서 test error가 더 낮은 모델과 높은 모델을 비교했는데 test error가 높은 모델의 norm이 더 작았음)
